pkg/devfile/parser/parse.go

```diff
@@ -212,11 +250,36 @@ func parseFromURI(uri string, curDevfileCtx devfileCtx.DevfileCtx) (DevfileObj,
 		d.Ctx = devfileCtx.NewURLDevfileCtx(u.String())
 	}
 	d.Ctx.SetURIMap(curDevfileCtx.GetURIMap())
-	// Fill the fields of DevfileCtx struct
-	err = d.Ctx.PopulateFromURL()
-	if err != nil {
-		return DevfileObj{}, err
+	return populateAndParseDevfile(d, true)
+}
+
+func parseFromRegistry(registryId, registryURL string, curDevfileCtx devfileCtx.DevfileCtx) (DevfileObj, error) {
+	if registryURL != "" {
+		devfileContent, err := getDevfileFromRegistry(registryId, registryURL)
+		if err != nil {
+			return DevfileObj{}, err
+		}
+		return ParseDevfile(ParserArgs{data: devfileContent, registryURLs: curDevfileCtx.GetRegistryURLs()})
+	} else if curDevfileCtx.GetRegistryURLs() != nil {
+		for _, registry := range curDevfileCtx.GetRegistryURLs() {
+			devfileContent, err := getDevfileFromRegistry(registryId, registry)
+			if devfileContent != nil && err == nil {
```

Comment:
I think we should show/demo this to Elson, idk about returning the first match. 

If you think about it 🤔 
1. order is not always guaranteed between registry URLs
2. devfile id in different registry can have different contents

Reply:
good point. I've never thought about multiple matches
we should discuss that in our meeting

Reply:
IIRC odo has a default registry or a flag is provided to download devfile from that registry during odo create

So I think we should have only one instead of an array

Reply:
confirmed with Elson, it should be an arry of registry URLs. and should return the first match if multiple occurrence presents